webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlResultTabsService.ts

```diff
@@ -62,8 +64,10 @@ export class SqlResultTabsService {
     return true;
   }
 
-  removeResultTabs(state: ISqlEditorTabState): void {
-    for (const tab of state.tabs.slice()) {
+  removeResultTabs(state: ISqlEditorTabState, excludedTabIds?: string[]): void {
+    const notExcludedTabs = state.tabs.filter(tab => !excludedTabIds?.includes(tab.id));
+
+    for (const tab of notExcludedTabs) {
       this.removeTab(state, tab);
     }
   }
```

Comment:
When you already have `for` you can use `if` to prevent "double work"
```suggestion
    for (const tab of state.tabs) {
      if (excludedTabIds?.includes(tab.id)) {
        continue;
      }
      this.removeTab(state, tab);
    }
  }
```